src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h

```diff
@@ -131,14 +131,14 @@ class GenericThreadStackManagerImpl_OpenThread
         struct Service
         {
             otSrpClientService mService;
-            char mInstanceName[kMaxInstanceNameSize];
-            char mName[kMaxNameSize];
+            char mInstanceName[kMaxInstanceNameSize + 1];
+            char mName[kMaxNameSize + 1];
```

Comment:
Here i think mName already has the NULL implicitly in kMaxNameSize
`static constexpr uint8_t kMaxNameSize         = chip::Mdns::kMdnsTypeMaxSize + chip::Mdns::kMdnsProtocolTextMaxSize + 1;`

Just below kMaxHostNameSize is a magic number 32 ? 
`static constexpr uint8_t kMaxHostNameSize     = 32;` Are we sure its needsa +1 for the Null character or is int already included in the + 32. I think the only verify that was falling is the one testing mInstanceName as that one didn't include the Null char
`static constexpr uint8_t kMaxInstanceNameSize = chip::Mdns::kMdnsNameMaxSize;`